symphony.rb

```diff
@@ -0,0 +1,79 @@
+class Symphony < Formula
+  desc "Framework for solving mixed integer linear programs"
+  homepage "https://projects.coin-or.org/SYMPHONY"
+  url "http://www.coin-or.org/download/pkgsource/SYMPHONY/SYMPHONY-5.6.13.tgz"
+  sha256 "62353422dae24c24fe02b41ba9ffeecbcda2940b68cf477c2867df0aebaa6812"
+  head "https://projects.coin-or.org/svn/SYMPHONY/trunk"
+
+  depends_on "homebrew/science/asl" => :optional
+  depends_on "homebrew/science/glpk" => :optional
+  depends_on "homebrew/science/openblas" => :optional
+
+  option "without-openmp", "Disable openmp support"
+
+  asl_dep = (build.with? "asl") ? ["with-asl"] : []
+  openblas_dep = (build.with? "openblas") ? ["with-openblas"] : []
+  option "with-gmpl", "GNU Modeling Language support via GLPK"
+
+  depends_on "homebrew/science/mumps" => [:optional, "without-mpi"] + openblas_dep
+  depends_on "homebrew/science/suite-sparse" => [:optional] + openblas_dep
+
+  mumps_dep = (build.with? "mumps") ? ["with-mumps"] : []
+  suite_sparse_dep = (build.with? "suite-sparse") ? ["with-suite-sparse"] : []
+
+  depends_on "mysql" => :build if build.with? "gmpl"
+  depends_on "readline" => :recommended
+
+  depends_on "clp" => (asl_dep + openblas_dep + mumps_dep + suite_sparse_dep)
+  depends_on "cgl"
+  depends_on :fortran
+
+  def install
+    args = ["--disable-debug",
+            "--disable-dependency-tracking",
+            "--prefix=#{prefix}",
+            "--datadir=#{pkgshare}",
+            "--includedir=#{include}/cbc",
+            "--with-sample-datadir=#{Formula["coin_data_sample"].opt_pkgshare}/coin/Data/Sample",
+            "--with-netlib-datadir=#{Formula["coin_data_netlib"].opt_pkgshare}/coin/Data/Netlib",
+            "--with-dot",
+	    "--enable-gnu-packages",
+	    "--with-application",
+           ]
+
+    if build.with? "readline"
+      ENV.append "CXXFLAGS", "-I#{Formula["readline"].opt_include}"
+      ENV.append "LDFLAGS",  "-L#{Formula["readline"].opt_lib} -lreadline"
+    end
+
+    if build.with? "gmpl"
+      # Symphony uses a patched version of GLPK for reading MPL files.
+      # Use a private version rather than require the Homebrew version of GLPK.
+      cd "ThirdParty/Glpk" do
+        system "./get.Glpk"
+      end
```

Comment:
I don't really know. SYMPHONY is only built in testing against the version of Glpk specified in the script and the patch also applies to that version specifically. I guess we could try to use the official version and just fix problems as they arise. I have no recollection at all of what the patch was needed for. I will play with it a bit and see, but is this something we could live with for now and address later? I don't want to hold up development of the remaining recipes and this might take a bit of time.

Actually, this prompted to go back and look at what we're doing with Osi. There, we attempt to use the official version, but it actually won't work, since we don't pass any arguments to configure to get it link with GLPK. We need to fix that in some fashion and if we decide to link to the official project there, then we need to see if that causes problems, too. As with SYMPHONY, OsiGlpk is also only made to work with a particular version of GLPK and I think there is a compelling argument that we need to link to link to that specific version of Glpk, as it's not updated frequently (if at all). If we link to specific version there, I'm not sure we can/should do it differently in other projects. I'll open an issue about this.